### PR TITLE
Instructors can save or publish questions

### PR DESCRIPTION
**Pull request by Bikranta**
In this commit, Instructor are allowed to either save or publish questions. if instructor save the question, only other instructor can view the question. student can view the question only if instructor publish the question. the published date and time is noted and shown in the question tab.

## Changes 
- Added an ability to Publish a question. 
## Context 
- When Instructor is adding the question 

## Testing 
- Create a question and save it. Check published date (should be empty because it is not published yet)
- Create a question and publish it. Check the published date
- Edit the saved question and publish it. Check the published date

## Pre-merge checklist 

- [x] Build succeeds locally and in CI
- [x] Proper documentation
- [x] Tests
- [x] Git history clean

### CODE REVIEW

File: app/controllers/questions_controller.rb [hunk 1]
```diff
@@ -28,6 +28,7 @@ def short
     @question.build_correct_answer
   end
 
+
```
Comment (strburst) [inline app/controllers/questions_controller.rb]: I noticed you added a bunch of extra line-breaks in different places—were they all intended?

File: app/controllers/questions_controller.rb [hunk 2]
```diff
@@ -36,12 +37,14 @@ def create
...
     end
+    @question.published_at = Time.zone.now if publishing?
 
     respond_to do |format|
       if @question.save
+        @question.published_at = Time.zone.now if publishing?
         format.html { redirect_to @question, notice: 'Question was successfully created.' }
         format.json { render :show, status: :created, location: @question }
-      else
+      else
```
Comment (strburst) [inline app/controllers/questions_controller.rb]: Get rid of this trailing whitespace. In the future, you can avoid this by making sure your editor is configured to trim trailing whitespace (or respect the `.editorconfig`).

File: app/views/questions/_form.html.haml [hunk 3]
```diff
@@ -35,7 +35,11 @@
       #new-answer{:style => "display: none;"}
         = render 'answer', f: g
 
+  #publish
```
Comment (strburst) [inline app/views/questions/_form.html.haml]: Is something in the form hooking onto this div?

**Review by strburst — CHANGES_REQUESTED** [review]
After testing this on my machine, it appears that hiding unpublished questions from students is still unimplemented. Is this a WIP? (Marking as "Request changes" if this is the case.)

I think it would probably be a good idea to add a "publish" button to the `show` action also, but we can do that later.

After that, a few style comments.

File: app/controllers/questions_controller.rb (hunk 1 above)
Comment (Bikranta) [reply app/controllers/questions_controller.rb]: i can edit those. thanks

File: app/controllers/questions_controller.rb (hunk 2 above)
Comment (Bikranta) [reply app/controllers/questions_controller.rb]: got it!

**Comment by Bikranta** [comment]
@strburst, hiding unpublished question from student isn't implemented because the student can't answer the question right now. when we implement how the students see the question, i thought we can add the visible part there.

**Review by strburst — CHANGES_REQUESTED** [review]
If you can, rebase the small commits that fixed style issues into the larger commit with the `fixup` option. Also, make the first line of that commit with the long message shorter and just continue the message on later paragraphs (see http://chris.beams.io/posts/git-commit/).

Noting this as a future feature: it would be nice if times displayed in the user's time zone (https://robots.thoughtbot.com/its-about-time-zones).

**Review by strburst — APPROVED** [review]

**Pull request merged**